# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/jsapi/RTCRtpReceiver.h
# Commit: 99ea4f221834
# Full Hash: 99ea4f221834f706be3738a92183287b54d8f718
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2022-12-22 17:00:32
# Regressor Bug: 1800136
# File Overlap Count: 1
# Description:
#   Bug 1800136: Make RTCRtpSender::GetStats get ICE stats. r=ng
#   
#   Also fix bugs where we would get duplicate ICE stats. This mostly mattered to
#   WebrtcGlobalInformation since RTCStatsReportInternal does not automatically
#   remove duplicate ids like RTCStatsReport (RTCStatsReport is maplike).
# ==============================================================================

diff -r 5801060dd3d0 -r 99ea4f221834 dom/media/webrtc/jsapi/RTCRtpReceiver.h
--- a/dom/media/webrtc/jsapi/RTCRtpReceiver.h	Thu Dec 22 02:33:56 2022 +0000
+++ b/dom/media/webrtc/jsapi/RTCRtpReceiver.h	Thu Dec 22 02:33:57 2022 +0000
@@ -66,7 +66,8 @@
       const uint32_t aRtpTimestamp, const bool aHasLevel, const uint8_t aLevel);
 
   nsPIDOMWindowInner* GetParentObject() const;
-  nsTArray<RefPtr<RTCStatsPromise>> GetStatsInternal();
+  nsTArray<RefPtr<RTCStatsPromise>> GetStatsInternal(
+      bool aSkipIceStats = false);
 
   void Shutdown();
   void BreakCycles();